tests/cpu_cortexm_address_check/main.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2018 Unwired Devices LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for Cortex-M address checks
+ *
+ * @author      Oleg Artamonov <oleg@unwds.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cpu.h"
+#include "shell.h"
+
+static int cmd_check(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <hex address>\n", argv[0]);
+        printf("\t example: %s 0x08080000\n", argv[0]);
+        return 1;
+    }
+
+    char *address;
+    uint32_t address_value;
+
+    sscanf(argv[1] + 2, "%lx", &address_value); /* ignore 0x prefix */
```

Comment:
And some code to check if the parse is successful.

Reply:
It can be done, but just leave it, it's a test after all. The important thing is that it does not blow up and that is already fixed.

BTW, [here](https://linux.die.net/man/3/strtol) is how to do it:

> In particular, if *nptr is not '\0' but **endptr is '\0' on return, the entire string is valid.